src/services/meta_transaction_service.ts

```diff
@@ -109,6 +166,79 @@ export class MetaTransactionService implements IMetaTransactionService {
         };
     }
 
+    /**
+     * Internal function to get meta-transaction v2 quote. The function is currently a copy of (with minor modifications) `_getMetaTransactionQuoteAsync` for scaffolding.
+     */
+    private async _getMetaTransactionV2QuoteAsync(
+        params: MetaTransactionV2QuoteParams,
+        endpoint: 'price' | 'quote',
+    ): Promise<MetaTransactionV2QuoteResult> {
+        const wrappedNativeToken = NATIVE_FEE_TOKEN_BY_CHAIN_ID[CHAIN_ID];
+
+        const quoteParams = {
+            ...params,
+            // NOTE: Internally all ETH trades are for WETH, we just wrap/unwrap automatically
+            buyToken: params.isETHBuy ? wrappedNativeToken : params.buyTokenAddress,
+            endpoint,
+            isMetaTransaction: true,
+            isUnwrap: false,
+            isWrap: false,
+            sellToken: params.sellTokenAddress,
+            shouldSellEntireBalance: false,
+            skipValidation: true,
+        };
+
+        const quote = await this._swapService.calculateSwapQuoteAsync(quoteParams);
+
+        // Quote Report
+        if (endpoint === 'quote' && quote.extendedQuoteReportSources && kafkaProducer) {
+            const quoteId = getQuoteIdFromSwapQuote(quote);
+            publishQuoteReport(
```

Comment:
should we publish fee info to kafka in the quote report?

Reply:
Yep. We should add it after the fee calculation & fee storage code are completed. For scaffolding we can't add it now as the necessary code has been committed yet